src/app/clusters/basic/basic.cpp

```diff
@@ -62,41 +62,251 @@ CHIP_ERROR BasicAttrAccess::Read(const ConcreteReadAttributePath & aPath, Attrib
+        char productLabel[kMaxLen + 1] = { 0 };
+        status                         = ConfigurationMgr().GetProductLabel(productLabel, sizeof(productLabel));
+
+        // TODO: Remove defaulting once proper runtime defaulting of unimplemented factory data is done
+        if (status == CHIP_DEVICE_ERROR_CONFIG_NOT_FOUND || status == CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE)
+        {
+            productLabel[0] = '\0';
+            status          = CHIP_NO_ERROR;
+        }
+
+        if (status == CHIP_NO_ERROR)
+        {
+            status = aEncoder.Encode(chip::CharSpan(productLabel, strnlen(productLabel, kMaxLen)));
+        }
+        break;
+    }
+
+    case SerialNumber::Id: {
+        constexpr size_t kMaxLen             = DeviceLayer::ConfigurationManager::kMaxSerialNumberLength;
+        char serialNumberString[kMaxLen + 1] = { 0 };
+        status                               = ConfigurationMgr().GetSerialNumber(serialNumberString, sizeof(serialNumberString));
+
+        // TODO: Remove defaulting once proper runtime defaulting of unimplemented factory data is done
+        if (status == CHIP_DEVICE_ERROR_CONFIG_NOT_FOUND || status == CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE)
+        {
+            serialNumberString[0] = '\0';
+            status                = CHIP_NO_ERROR;
+        }
+
+        if (status == CHIP_NO_ERROR)
+        {
+            status = aEncoder.Encode(chip::CharSpan(serialNumberString, strnlen(serialNumberString, kMaxLen)));
+        }
+        break;
+    }
+
+    case UniqueID::Id: {
+        constexpr size_t kMaxLen   = DeviceLayer::ConfigurationManager::kMaxUniqueIDLength;
+        char uniqueId[kMaxLen + 1] = { 0 };
+        status                     = ConfigurationMgr().GetUniqueId(uniqueId, sizeof(uniqueId));
+
+        // TODO: Remove defaulting once proper runtime defaulting of unimplemented factory data is done
+        if (status == CHIP_DEVICE_ERROR_CONFIG_NOT_FOUND || status == CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE)
         {
-            err = aEncoder.Encode(chip::CharSpan("XX", strlen("XX")));
+            uniqueId[0] = '\0';
+            status      = CHIP_NO_ERROR;
         }
-        else
+
+        if (status == CHIP_NO_ERROR)
         {
-            err = aEncoder.Encode(chip::CharSpan(location, strlen(location)));
+            status = aEncoder.Encode(chip::CharSpan(uniqueId, strnlen(uniqueId, kMaxLen)));
         }
+        break;
+    }
+
+    default:
+        // We did not find a processing path, the caller will delegate elsewhere.
+        status = CHIP_ERROR_NOT_FOUND;
+        break;
     }
-    else
+
+    if (status == CHIP_ERROR_NOT_FOUND)
     {
-        err = aEncoder.Encode(chip::CharSpan("XX", strlen("XX")));
+        // Somehow attribute support not handled by custom logic. Until all such
+        // paths are resolved, allow fallback to attribute store or other heuristics
+        // which may be done by callers of this method.
+        status = CHIP_NO_ERROR;
     }
```

Comment:
I thought we had a concept of handled / not handled, which then I thought could fallback to some other handler (attribute store?) for not-handled.  Can something like that be leveraged here to clean this up?  Or is that in fact what we are already doing?  If so, why do we need to translate here from CHIP_ERROR_NOT_FOUND to CHIP_NO_ERROR?  Why can't the path where something is not found just return CHIP_NO_ERROR if we intend that this is not an error?